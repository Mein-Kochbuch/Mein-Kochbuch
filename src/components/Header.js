import React from "react";
import {StyleSheet, Text, View} from "react-native";

export default function Header({title}) {

    return (
        <View style={styles.container}>
            <Text style={styles.heading}>
                {title}
            </Text>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        marginVertical: 12,
        display: "flex",
        justifyContent: "center",
    },
    heading: {
        textAlign: "center",
        fontSize: 50,
    }
})
