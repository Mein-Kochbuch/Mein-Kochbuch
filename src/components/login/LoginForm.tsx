import React, {useState} from "react";
import {Linking} from "react-native";
import InputField from "./InputField";
import styled from "styled-components/native/dist/styled-components.native.esm";
import {Colors} from "react-native/Libraries/NewAppScreen";
import {useColorScheme} from "react-native-appearance";

export default function LoginForm({onSubmit}) {
    const [credentials, setCredentials] = useState({})

    const isDarkMode = useColorScheme() === 'dark';

    const onEmailChange = (value) => {
        setCredentials((oldCredentials) => {
            return {...oldCredentials, "username": value}
        })
    }

    const onPasswordChange = (value) => {
        setCredentials((oldCredentials) => {
            return {...oldCredentials, "password": value}
        })
    }

    return (
        <StyledView>
            <InputField
                title={"E-Mail"}
                onChangeText={onEmailChange}
                placeholder="E-Mail"
                value={credentials.username}
                keyboardType={"email-address"}/>
            <InputField
                title={"Password"}
                onChangeText={onPasswordChange}
                placeholder="Password"
                value={credentials.password}
                secureTextEntry={true}/>

            <PrivacyBanner>
                Wenn du fortfährst, stimmst du den
                <LinkText onPress={() => Linking.openURL('https://mein-kochbuch.org/agb/')}> AGBs </LinkText>
                zu. In unseren
                <LinkText onPress={() => Linking.openURL('https://mein-kochbuch.org/privacypolicy/')}> Privacy
                    Policy </LinkText>
                erfährst du, wie wir deine Daten erfassen, verwenden und teilen.
            </PrivacyBanner>

            <StyledTouchableOpacity isDarkMode={isDarkMode} onPress={() => onSubmit(credentials)}>
                <StyledText>
                    Login
                </StyledText>
            </StyledTouchableOpacity>
        </StyledView>
    )
}

const StyledView = styled.View`
  display: flex;
  align-items: center;
  flex-direction: column;
`

const LinkText = styled.Text`
  color: #0074D9;
  text-decoration-line: underline;
`

const PrivacyBanner = styled.Text`
  margin: 24px;
  text-align: center;
`

const StyledText = styled.Text`
  text-align: center;
  font-size: 30px;
`

const StyledTouchableOpacity = styled.TouchableOpacity`
  margin: 12px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  width: 80%;
  min-width: 300px;
  border-style: solid;
  border-color: ${props => props.isDarkMode ? Colors.lighter : Colors.darker};
  border-radius: 10px;
  border-width: 2px;
`

