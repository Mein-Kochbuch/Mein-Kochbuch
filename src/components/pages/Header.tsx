import React from "react";
import {StyleSheet, Text, View} from "react-native";

interface HeaderProps {
    title: string
}

export default function Header({title}: HeaderProps) {

    return (
        <View style={styles.container}>
            <Text style={styles.heading}>
                {title}
            </Text>
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        marginVertical: 12,
        display: "flex",
        justifyContent: "center",
    },
    heading: {
        textAlign: "center",
        fontSize: 50,
    }
})
