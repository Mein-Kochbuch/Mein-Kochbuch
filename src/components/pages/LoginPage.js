import React, {useContext, useState} from "react";
import {View} from "react-native";
import Header from "./Header";
import styled from "styled-components/native";
import LoginForm from "../login/LoginForm";
import {Colors} from "react-native/Libraries/NewAppScreen";
import {useColorScheme} from "react-native-appearance";
import {AuthContext} from "../../context/AuthProvider";
import ErrorText from "../../utils/ErrorText";

export default function LoginPage() {

    const [loginError, setLoginError] = useState()
    const isDarkMode = useColorScheme() === 'dark';
    const {login} = useContext(AuthContext)

    const onSubmit = (credentials) => {
        login(credentials)
            .catch((error) => {
                console.log(error)
                setLoginError(error.toString())
            })
    }


    return (
        <StyledView>
            <Header title={"Login"}/>
            <ErrorText text={loginError}/>
            <LoginForm onSubmit={onSubmit}/>
            <View>
                <RegisterButton isDarkMode={isDarkMode} onPress={() => {
                }}>
                    <StyledText>
                        Register
                    </StyledText>
                </RegisterButton>
            </View>
        </StyledView>
    )
}

const StyledView = styled.View`
  display: flex;
  align-items: center;
  flex-direction: column;
`

const StyledText = styled.Text`
  text-align: center;
  font-size: 30px;
`

const RegisterButton = styled.TouchableOpacity`
  margin: 12px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  width: 80%;
  min-width: 300px;
  border-style: solid;
  border-color: ${props => props.isDarkMode ? Colors.lighter : Colors.darker};
  border-radius: 10px;
  border-width: 2px;
`
