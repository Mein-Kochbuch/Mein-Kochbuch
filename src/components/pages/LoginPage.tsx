import React, {useContext, useState} from "react";
import {View} from "react-native";
import Header from "./Header";
import styled from "styled-components/native";
import LoginForm from "../login/LoginForm";
import {AuthContext} from "../../context/AuthProvider";
import ErrorText from "../../utils/ErrorText";
import RegisterButton from "../login/RegisterButton";

export default function LoginPage() {

    const [loginError, setLoginError] = useState<string>()
    const {login} = useContext(AuthContext)

    const onSubmit = (credentials: { username: string, password: string }) => {
        login(credentials)
            .catch((error: { toString: () => string }) => {
                console.log(error)
                setLoginError(error.toString())
            })
    }


    return (
        <StyledView>
            <Header title={"Login"}/>
            {loginError && <ErrorText text={loginError}/>}
            <LoginForm onSubmit={onSubmit}/>
            <View>
                <RegisterButton/>
            </View>
        </StyledView>
    )
}

const StyledView = styled.View`
  display: flex;
  align-items: center;
  flex-direction: column;
`


