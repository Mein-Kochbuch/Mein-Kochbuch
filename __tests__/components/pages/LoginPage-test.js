import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import Header from '../../../src/components/pages/Header';
import LoginPage from '../../../src/components/pages/LoginPage';
import LoginForm from '../../../src/components/login/LoginForm';

jest.mock('../../../src/components/pages/Header', () => {
  return props => {
    return <div {...props} />;
  };
});

jest.mock('../../../src/utils/ErrorText', () => {
  return props => {
    return <div {...props} />;
  };
});

jest.mock('../../../src/components/login/LoginForm', () => {
  return props => {
    return <div {...props} />;
  };
});

jest.mock('../../../src/components/login/RegisterButton', () => {
  return props => {
    return <div {...props} />;
  };
});

it('LoginPage Test', () => {
  const component = renderer.create(<LoginPage />);

  expect(component.toJSON()).toMatchSnapshot();
  expect(component.toJSON().children.length).toBe(3);
  const testInstance = component.root;
  expect(testInstance.findByType(Header).props.title).toBe('Login');
  expect(typeof testInstance.findByType(LoginForm).props.onSubmit).toBe(
    'function',
  );
});
