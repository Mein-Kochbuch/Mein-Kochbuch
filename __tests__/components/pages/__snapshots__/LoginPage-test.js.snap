// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage Test 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "column",
      },
    ]
  }
>
  <div
    title="Login"
  />
  <div
    onSubmit={[Function]}
  />
  <View>
    <div />
  </View>
</View>
`;
